### PR TITLE
 Datasource abstraction

### PR DESCRIPTION
**Pull request by bharathappali**
This PR adds the datasource abstraction for the autotune datasources.

PR holds only the interfaces and the respective utils and exceptions added.

**Interfaces:**
```
AutotuneDatasource.java
AutotuneDatasourceCollection.java
AutotuneDatasourceServiceability.java
```

**Exceptions:**
```
AutotuneDatasourceDoesNotExist.java
AutotuneDatasourceAlreadyExists.java
```

**Utilities:**
```
CommonUtils.java
```

All the files are added to `com.autotune.common` package

Signed-off-by: bharathappali <abharath@redhat.com>

### CODE REVIEW

**Comment by bharathappali** [comment]
@dinogun Can I please get it reviewed?

File: src/main/java/com/autotune/common/data/datasource/AutotuneDatasource.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * Currently Supported Implementations:
+ *  - Prometheus
+ *
+ *  The Implementation should maintain 3 variables : Name, Type and Source and
+ *  should be set or get using these API's
+ *
+ *  var name:   Type String
+ *  var type:   Type AutotuneDatasourceTypes
+ *  var source: Type String
```
Comment (dinogun) [inline src/main/java/com/autotune/common/data/datasource/AutotuneDatasource.java:33]: Add the actual values that we use in Prometheus here.

File: src/main/java/com/autotune/common/utils/CommonUtils.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * DatasourceReliabilityStatus is a ENUM which holds the possible statuses
+     * to return for checking if a datasource is reliable
+     *
+     * RELIABLE - states that the datasource is reliable
+     * NOT_RELIABLE -  states that the datasource is not reliable
+     */
+    public enum DatasourceReliabilityStatus {
+        RELIABLE,
+        NOT_RELIABLE,
```
Comment (dinogun) [inline src/main/java/com/autotune/common/utils/CommonUtils.java:69]: We need specific examples of when we use these two values

File: src/main/java/com/autotune/common/data/datasource/AutotuneDatasource.java (hunk 1 above)
Comment (bharathappali) [reply src/main/java/com/autotune/common/data/datasource/AutotuneDatasource.java:33]: Done

File: src/main/java/com/autotune/common/utils/CommonUtils.java (hunk 2 above)
Comment (bharathappali) [reply src/main/java/com/autotune/common/utils/CommonUtils.java]: Done

**Pull request merged**